### 파일 read/open

In [1]:
%pwd

'c:\\Source\\kdt-python-2025\\chapter13'

### os 모듈 + 문자열 기반 코드



In [2]:
import os

# 경로 조합
dir_name = "data"
file_name = "pi_digits.txt"
path = os.path.join(dir_name, file_name)

# 파일 읽기
with open(path, "r", encoding="utf-8") as f:
    contents = f.read()
print("[os] 읽은 내용:", repr(contents))

# 파일 쓰기
new_path = os.path.join(dir_name, "pi_digits_copy.txt")
with open(new_path, "w", encoding="utf-8") as f:
    f.write(contents)
print("[os] 복사 완료:", new_path)


[os] 읽은 내용: '3.1415926535\n  8979323846\n  2643383279\n'
[os] 복사 완료: data\pi_digits_copy.txt


- pathlib 모듈은 객체 지향적인 방식으로 파일 시스템 경로를 다룬다

- 기존의 with open(filename, 'r', encoding='utf-8') as f: f.read() 방식보다 훨씬 간결

pathlib 모듈은 파일과 디렉터리 경로를 객체로 Python 표준 라이브러리

Path는 이 모듈 내의 클래스로, 경로 객체(path object)를 생성

In [3]:
from pathlib import Path

path = Path('data\pi_digits.txt')
contents = path.read_text() 
print(repr(contents) + '▶')
print(contents) 
#contents2 = path.read_text().strip() ## strip()은 전체 문자열 시작과 끝의 공백은 제거
contents = contents.rstrip()
print(repr(contents)) # 결과에서 3줄, 4줄 확인 
print(contents + '▶')

'3.1415926535\n  8979323846\n  2643383279\n'▶
3.1415926535
  8979323846
  2643383279

'3.1415926535\n  8979323846\n  2643383279'
3.1415926535
  8979323846
  2643383279▶


<>:3: SyntaxWarning: invalid escape sequence '\p'
<>:3: SyntaxWarning: invalid escape sequence '\p'
C:\Users\user\AppData\Local\Temp\ipykernel_1872\988804548.py:3: SyntaxWarning: invalid escape sequence '\p'
  path = Path('data\pi_digits.txt')


In [7]:
new_lines = [line.strip() for line in contents.splitlines()]
print(new_lines)

['3.1415926535', '8979323846', '2643383279']


In [8]:
import os
print(os.getcwd())

c:\Users\user\OneDrive\소스코드\파이썬소스\python_source\chap13


pi = 원둘레 / 지름 = 3.14159... /1

원둘레 = 2pi r (r은 반지름)

###중2 코딩 이야기###

1. 실험 공간: 정사각형 (네모)

가로: 2, 세로: 2

좌표: x∈[−1,1], y∈[−1,1]

면적: 2×2=4

✅ 2. 원의 조건
중심: (0, 0)

반지름: 1

원의 면적: pi * r*r = pi

비율 = 원면적/네모 면적 = pi/4
 
 (원에 있는 점의 수)/(전체 점의 수) = pi/4

In [8]:
import random

def estimate_pi(n_points):
    inside_circle = 0
    for _ in range(n_points):
        x = random.uniform(-1, 1)
        y = random.uniform(-1, 1)
        if x**2 + y**2 <= 1:
            inside_circle += 1
    return 4 * inside_circle / n_points

print("근사한 π 값 =", estimate_pi(1000)) # 백만개, 천만개, 억개로 증가 시켜서 계산 


근사한 π 값 = 3.028


In [11]:
# n_points = 1000 ~ 10000 까지 1000씩 증가
for n in range(1000, 100001, 1000):
    pi_est = estimate_pi(n)
    print(f"n_points={n:5d} → 근사 π = {pi_est}")

n_points= 1000 → 근사 π = 3.144
n_points= 2000 → 근사 π = 3.102
n_points= 3000 → 근사 π = 3.1706666666666665
n_points= 4000 → 근사 π = 3.144
n_points= 5000 → 근사 π = 3.1544
n_points= 6000 → 근사 π = 3.1553333333333335
n_points= 7000 → 근사 π = 3.111428571428571
n_points= 8000 → 근사 π = 3.1515
n_points= 9000 → 근사 π = 3.1515555555555554
n_points=10000 → 근사 π = 3.1268
n_points=11000 → 근사 π = 3.1563636363636363
n_points=12000 → 근사 π = 3.1203333333333334
n_points=13000 → 근사 π = 3.1344615384615384
n_points=14000 → 근사 π = 3.1517142857142857
n_points=15000 → 근사 π = 3.138933333333333
n_points=16000 → 근사 π = 3.1305
n_points=17000 → 근사 π = 3.1461176470588237
n_points=18000 → 근사 π = 3.1426666666666665
n_points=19000 → 근사 π = 3.128
n_points=20000 → 근사 π = 3.1526
n_points=21000 → 근사 π = 3.1283809523809523
n_points=22000 → 근사 π = 3.1683636363636363
n_points=23000 → 근사 π = 3.1502608695652174
n_points=24000 → 근사 π = 3.146166666666667
n_points=25000 → 근사 π = 3.132
n_points=26000 → 근사 π = 3.146923076923077
n_points=27

In [20]:
from pathlib import Path
path = Path('data\pi_million_digits.txt')
contents = path.read_text()

print(contents)

3.1415926535897932384626433832795028841971693993751058209749445923078164062862089986280348253421170679
  8214808651328230664709384460955058223172535940812848111745028410270193852110555964462294895493038196
  4428810975665933446128475648233786783165271201909145648566923460348610454326648213393607260249141273
  7245870066063155881748815209209628292540917153643678925903600113305305488204665213841469519415116094
  3305727036575959195309218611738193261179310511854807446237996274956735188575272489122793818301194912
  9833673362440656643086021394946395224737190702179860943702770539217176293176752384674818467669405132
  0005681271452635608277857713427577896091736371787214684409012249534301465495853710507922796892589235
  4201995611212902196086403441815981362977477130996051870721134999999837297804995105973173281609631859
  5024459455346908302642522308253344685035261931188171010003137838752886587533208381420617177669147303
  59825349042875546873115956286388235378759375195778185778053217122680661

<>:2: SyntaxWarning: invalid escape sequence '\p'
<>:2: SyntaxWarning: invalid escape sequence '\p'
C:\Users\user\AppData\Local\Temp\ipykernel_1872\1792253135.py:2: SyntaxWarning: invalid escape sequence '\p'
  path = Path('data\pi_million_digits.txt')


In [21]:
contents = path.read_text()
print(repr(contents))


'3.1415926535897932384626433832795028841971693993751058209749445923078164062862089986280348253421170679\n  8214808651328230664709384460955058223172535940812848111745028410270193852110555964462294895493038196\n  4428810975665933446128475648233786783165271201909145648566923460348610454326648213393607260249141273\n  7245870066063155881748815209209628292540917153643678925903600113305305488204665213841469519415116094\n  3305727036575959195309218611738193261179310511854807446237996274956735188575272489122793818301194912\n  9833673362440656643086021394946395224737190702179860943702770539217176293176752384674818467669405132\n  0005681271452635608277857713427577896091736371787214684409012249534301465495853710507922796892589235\n  4201995611212902196086403441815981362977477130996051870721134999999837297804995105973173281609631859\n  5024459455346908302642522308253344685035261931188171010003137838752886587533208381420617177669147303\n  5982534904287554687311595628638823537875937519577818577805321

In [22]:
lines = contents.splitlines() #줄 단위로 나누어 리스트를 반환
print(lines[0])
print(lines[1])

3.1415926535897932384626433832795028841971693993751058209749445923078164062862089986280348253421170679
  8214808651328230664709384460955058223172535940812848111745028410270193852110555964462294895493038196


In [23]:
pi_string = ''
for line in lines:
    pi_string += line.strip()
    
birthday = input("생년월일(mmddyy): ")
if (birthday in pi_string):
    print("마이 생일 있다", birthday)
else:
    print('없다')

마이 생일 있다 031401


open() 함수: 파일을 열어주는 내장 함수
with 구문: 파일을 열고 나서 자동으로 닫히게 해주는 안전한 문법 > file.close()를 따로 쓸 필요가 없다

In [21]:
with open('data\pi_digits.txt', 'r') as file:
    data = file.read()
    print(data)
print(data)

3.1415926535
  8979323846
  2643383279

3.1415926535
  8979323846
  2643383279



In [25]:
import os #os는 Python의 표준 라이브러리 모듈

# 현재 디렉터리의 파일 및 폴더 목록 출력
current_directory = os.listdir('.') # '.'는 **현재 디렉터리(current directory)**를 의미, 결과는 리스트
print("현재 디렉터리의 파일 목록:", current_directory)

# 특정 디렉터리의 파일 목록 출력
directory_path = 'data'  # 경로는 환경에 맞게 설정
files_in_directory = os.listdir(directory_path)
print(f"{directory_path}의 파일 목록:", files_in_directory)
for fl in files_in_directory:
    print(fl)

현재 디렉터리의 파일 목록: ['13장_파일.ipynb', '13장_파일실습_교재외.ipynb', 'binfile.bin', 'config.dat', 'data', 'gist1.py', 'gist2.py', 'hello.txt', 'list1301.py', 'list1301a.py', 'list1301b.py', 'list1302.py', 'list1303.py', 'list1304.py', 'list1304a.py', 'list1304b.py', 'list1304c.py', 'list1305.py', 'list1306.py', 'list1307.py', 'list1308.py', 'list1309.py', 'list1310.py', 'list13c01.py', '실습16_1파일_가시화_과제.ipynb', '실습16_1파일_가시화_날씨_v1.0.ipynb']
data의 파일 목록: ['pi_digits.txt', 'pi_digits_copy.txt', 'pi_million_digits.txt', 'weather_data']
pi_digits.txt
pi_digits_copy.txt
pi_million_digits.txt
weather_data


### 파일 저장하기 

In [24]:
from pathlib import Path
print(Path.cwd()) # os.getcwd()
path = Path('learning_file.txt')
path.write_text('good nice love') # 객체 버전 -> 간결한 코딩

c:\Source\kdt-python-2025\chapter13


14

In [25]:
read_contents = path.read_text()
print(read_contents)

good nice love


In [26]:
write_contents = 'java python\n'
write_contents += 'cpp javascript\n'
path.write_text(write_contents)

27

In [27]:
read_contents = path.read_text()
print(read_contents) #overwrite됨 

java python
cpp javascript



기존 파일 끝에 **추가 (append)**하려면:

path.write_text()가 아니라 open()과 'a' 모드를 사용

In [28]:
print(path)

learning_file.txt


In [29]:
with open(path, 'a') as f:  # 'a'는 append 모드
    f.write('타입스크립트\n')     # 파일 끝에 추가


In [30]:
read_contents = path.read_text()
print(read_contents)

java python
cpp javascript
타입스크립트



In [31]:
# 여러 줄의 텍스트를 파일에 쓰기 (writelines 사용)
lines = [
    "첫 번째 줄\n",
    "두 번째 줄\n",
    "세 번째 줄\n"
]

with open('example.txt', 'w', encoding='utf-8') as file:
    file.writelines(lines)  # 여러 줄을 한 번에 쓰기



In [32]:
with open('example.txt', 'r', encoding='utf-8') as file2:
    new_lines = file2.readlines()  # 여러 줄을 한 번에 읽기
    
print(new_lines) # new_lines 변수는 with 블록 밖에 사용 가능하다 

['첫 번째 줄\n', '두 번째 줄\n', '세 번째 줄\n']


In [33]:
import os

# 상위 폴더 경로와 하위 폴더 이름
parent_folder = 'machine_learning'
child_folder = os.path.join(parent_folder, 'pywork')  # 경로를 조합하여 하위 폴더 생성

# 하위 폴더 생성
if not os.path.exists(child_folder):  # 하위 폴더가 없을 때만 생성
    os.makedirs(child_folder)
    print(f"'{child_folder}' 폴더가 생성되었습니다.")
else:
    print(f"'{child_folder}' 폴더가 이미 존재합니다.")

'machine_learning\pywork' 폴더가 생성되었습니다.


### csv 파일 공부

파일을 생성하고 이를 읽어서 처리하는 클래스 작성 실습

In [34]:
# 여러 줄의 텍스트를 파일에 쓰기 (writelines 사용)
lines = [
    "홍길동, 20, A",
    "강감찬, 22, B",
    "계백, 21, C"
]

with open('students.txt', 'w', encoding='utf-8') as file:
    for line in lines:
        file.write(line + '\n')  # 각 줄 끝에 개행 문자 추가




In [35]:
with open('students.txt', 'r', encoding='utf-8') as file2:
    new_lines = file2.readlines()  # 여러 줄을 한 번에 읽기
    
print(new_lines) # new_lines 변수는 with 블록 밖에 사용 가능하다 

['홍길동, 20, A\n', '강감찬, 22, B\n', '계백, 21, C\n']


### read() / readline() / readlines() 와 비교

- file.read() > 파일 전체를 한번에 문자열로 읽음

- file.readlines() > 모든 줄을 리스트로 읽음

- file.readline() > 한 줄씩 읽음

- for line in file > 한 줄씩 자동 반복 

In [ ]:
# Student 클래스 정의
class Student:
    def __init__(self, name, age, grade):
        self.name = name
        self.age = age
        self.grade = grade

    def __str__(self):
        return f"이름: {self.name}, 나이: {self.age}, 학년: {self.grade}"

# 파일에서 데이터 읽기 및 Student 객체 생성
students = []

with open('students.txt', 'r', encoding='utf-8') as file:
    for line in file:
        # 각 줄의 데이터를 쉼표로 분리
        name, age, grade = line.strip().split(', ')
        # Student 객체 생성
        student = Student(name, int(age), grade)
        students.append(student)

# 학생 객체 출력
for student in students:
    print(student) # student는 객체 - __str__() 자동 호출

이름: 홍길동, 나이: 20, 학년: A
이름: 강감찬, 나이: 22, 학년: B
이름: 계백, 나이: 21, 학년: C


In [36]:
lines = [
    "홍길동, 20, A",
    "강감찬, 22, B",
    "계백, 21, C"
]

with open('students.csv', 'w', encoding='utf-8') as file:
    for line in lines:
        file.write(line + '\n')  # 각 줄 끝 개행


### TXT vs CSV 차이 설명

- csv 파일:
  
  1) comma-separated values,  표 형태(행, 열)로 표현, 구분자로 콤마 사용, csv 모듈 사용

  2) txt 파일: 일반 텍스트 파일, 행/열에 대한 구조 없음, 값 구분 방식 없음, 파이썬에서 문자열 처리로 직접 split 필요/line.strip().split(', ')

#머신러닝 학습을 위한 CSV 파일 읽기 → 객체화 → 전처리

1) 파일을 읽어 객체 리스트로 저장

2) 전처리: 평균 점수 계산

3) 머신러닝에 활용할 수 있도록 feature vector 준비

CSV 파일을 텍스트 파일로 보고 직접 줄 단위로 읽고 split()으로 분리하는 방식으로 처리 가능

open() 함수와 문자열로 직접 처리하는 방식

In [37]:
# students.csv 내용 예시:
# 홍길동,20,A
# 강감찬,22,B
# 이순신,21,A

students = []
with open("students.csv", "r", encoding="utf-8") as file:
    for line in file:
        name, age, grade = line.strip().split(",")
        students.append((name, int(age), grade))

print(students)


[('홍길동', 20, ' A'), ('강감찬', 22, ' B'), ('계백', 21, ' C')]


csv 모듈(객체 기반) 사용 방식 (위보다 좋음)

- CSV 파일 형식 처리를 위한 표준 라이브러리

In [46]:
import csv

students = []
with open("students.csv", "r", encoding="utf-8", newline="") as file:
    reader = csv.reader(file) # reader는 csv의 iterable 객체
    print(type(reader))
    for row in reader: # row는 list를 리턴
        print(type(row))
        name, age, grade = row # split 사용x
        students.append((name, int(age), grade))

print(students)


<class '_csv.reader'>
<class 'list'>
<class 'list'>
<class 'list'>
[('홍길동', 20, ' A'), ('강감찬', 22, ' B'), ('계백', 21, ' C')]


In [49]:
import csv

data = [
    ["홍길동", 20, "A"],
    ["강감찬", 22, "B"],
    ["이순신", 21, "A"],
    ["계백", 23, "C"],
    ["을지문덕", 25, "D"],
    ["양만춘", 26, "F"],
]

with open("output.csv", "w", encoding="utf-8", newline="") as file:
    writer = csv.writer(file)
    for row in data:
        writer.writerow(row)


Pandas로 CSV 파일 읽기

In [47]:
%pip install pandas

   ---------------------------------------- 0.0/11.0 MB ? eta -:--:--
   ------ --------------------------------- 1.8/11.0 MB 10.0 MB/s eta 0:00:01
   ---------------- ----------------------- 4.5/11.0 MB 11.2 MB/s eta 0:00:01
   ------------------------- -------------- 7.1/11.0 MB 11.2 MB/s eta 0:00:01
   ---------------------------------- ----- 9.4/11.0 MB 11.5 MB/s eta 0:00:01
   ---------------------------------------- 11.0/11.0 MB 11.1 MB/s  0:00:01

   ---------------------------------------- 0/3 [pytz]
   ------------- -------------------------- 1/3 [tzdata]
   ------------- -------------------------- 1/3 [tzdata]
   -------------------------- ------------- 2/3 [pandas]
   -------------------------- ------------- 2/3 [pandas]
   -------------------------- ------------- 2/3 [pandas]
   -------------------------- ------------- 2/3 [pandas]
   -------------------------- ------------- 2/3 [pandas]
   -------------------------- ------------- 2/3 [pandas]
   -------------------------- 

In [50]:
import pandas as pd

df = pd.read_csv("output.csv", encoding="utf-8", header=None)
print(df)


      0   1  2
0   홍길동  20  A
1   강감찬  22  B
2   이순신  21  A
3    계백  23  C
4  을지문덕  25  D
5   양만춘  26  F


In [ ]:
import pandas as pd

df = pd.read_csv(
    "output.csv",
    encoding="utf-8",
    header=None,
    names=["이름", "나이", "학점"] # 컬럼 부여
)
print(type(df)) # DataFrame - 테이블 형태의 소트 데이터
print(df)


<class 'pandas.core.frame.DataFrame'>
     이름  나이 학점
0   홍길동  20  A
1   강감찬  22  B
2   이순신  21  A
3    계백  23  C
4  을지문덕  25  D
5   양만춘  26  F


In [57]:
print(df[df["학점"] == "A"]) # 학점이 A인 학생만 보기
print('---------------------')
print(df["나이"].mean()) # 평균 나이
print('---------------------')
print(df.sort_values("나이", ascending=False)) # 나이순 정렬


    이름  나이 학점
0  홍길동  20  A
2  이순신  21  A
---------------------
22.833333333333332
---------------------
     이름  나이 학점
5   양만춘  26  F
4  을지문덕  25  D
3    계백  23  C
1   강감찬  22  B
2   이순신  21  A
0   홍길동  20  A


In [58]:
import pandas as pd

# ① Student 클래스 정의
class Student:
    def __init__(self, name, age, grade):
        self.name = name
        self.age = age
        self.grade = grade

    def __str__(self):
        return f"이름: {self.name}, 나이: {self.age}, 학점: {self.grade}"


# ② CSV 파일을 데이터프레임으로 읽기
df = pd.read_csv("output.csv", encoding="utf-8", header=None, names=["이름", "나이", "학점"])

# ③ 데이터프레임을 Student 객체 리스트로 변환
students = []
for _, row in df.iterrows():
    students.append(Student(row["이름"], int(row["나이"]), row["학점"]))


# ④ 나이를 기준으로 정렬 (내림차순: 나이가 많은 학생부터)
students_sorted = sorted(students, key=lambda s: s.age, reverse=True)

# ⑤ 나이가 3번째인 학생 출력 (즉, 정렬된 리스트의 인덱스 2)
if len(students_sorted) >= 3:
    third_student = students_sorted[2]
    print("나이가 3번째인 학생:")
    print(third_student)
else:
    print("학생 수가 3명 미만입니다.")


나이가 3번째인 학생:
이름: 계백, 나이: 23, 학점: C


파일 예외처리

In [22]:
# Student 클래스 정의
class Student:
    def __init__(self, name, age, grade):
        self.name = name
        self.age = age
        self.grade = grade

    def __str__(self):
        return f"이름: {self.name}, 나이: {self.age}, 학년: {self.grade}"

# 파일에서 데이터 읽기 및 Student 객체 생성
students = []

with open('students2.txt', 'r', encoding='utf-8') as file:
    for line in file:
        # 각 줄의 데이터를 쉼표로 분리
        name, age, grade = line.strip().split(', ')
        # Student 객체 생성
        student = Student(name, int(age), grade)
        students.append(student)

# 학생 객체 출력
for student in students:
    print(student)

FileNotFoundError: [Errno 2] No such file or directory: 'students2.txt'

### AI·데이터 분석에서는 파일 예외 처리(file exception handling) 가 매우 중요한 이유

- 데이터 분석은 외부 파일을 다룸 : CSV/JSON/엑셀/로그/이미지 등 다양한 파일 포맷을 다루기 때문에 오류가 발생할 확률이 높음

- 값 누락·열 부족·잘못된 형식이 있으면 잘못된 학습/결과 발생

- 모델이 잘못된 데이터를 학습하면 품질 하락, 재학습 비용 증가

### 실무에서 가장 많이 발생하는 파일 문제 유형

- 파일 없음 : 지정된 데이터셋 파일/경로가 존재하지 않음

- 잘못된 구분자 : 콤마 대신 탭·세미콜론

- 컬럼 불일치 : 헤더 개수가 맞지 않음, 누락

- 결측치 : NaN, NULL, 빈 문자열

- 데이터 타입 오류 : 숫자 → 문자열 / 날짜 파싱 실패

In [23]:
# Student 클래스 정의
class Student:
    def __init__(self, name, age, grade):
        self.name = name
        self.age = age
        self.grade = grade

    def __str__(self):
        return f"이름: {self.name}, 나이: {self.age}, 학년: {self.grade}"


students = []
filename = "students2.txt"

try:
    # 파일 열기 예외 처리
    with open(filename, 'r', encoding='utf-8') as file:
        for line_num, line in enumerate(file, start=1):
            try:
                # 공백 제거 및 필드 분리
                parts = line.strip().split(', ')
                if len(parts) != 3:
                    raise ValueError(f"필드 개수가 3개가 아님 → {parts}")

                name, age, grade = parts

                # 나이 정수 변환 예외 처리
                try:
                    age = int(age)
                except ValueError:
                    raise ValueError(f"나이가 정수가 아님 → '{age}'")

                # Student 객체 생성
                student = Student(name, age, grade)
                students.append(student)

            except Exception as e:
                print(f"[경고] {line_num}번째 줄에서 오류 발생: {e}")
                # 오류가 있어도 다음 줄 데이터 계속 처리

except FileNotFoundError:
    print(f"[오류] 파일을 찾을 수 없습니다: {filename}")

except Exception as e:
    print(f"[오류] 파일을 읽는 중 알 수 없는 에러: {e}")


# 정상적으로 읽힌 학생 객체 출력
print("\n=== 학생 정보 출력 ===")
for student in students:
    print(student)


[오류] 파일을 찾을 수 없습니다: students2.txt

=== 학생 정보 출력 ===


### 13장 파일 실습 과제1
- data.csv 파일 읽어 클래스 객체로 만들기


In [59]:
import csv
from datetime import datetime

# 1) 레코드를 담을 클래스 정의
class Record:
    def __init__(self, date_str, text):
        # date_str: "2021-01-01" 같은 문자열
        self.date_str = date_str
        self.text = text

        # 날짜를 datetime 객체로도 저장해두고 싶으면:
        try:
            self.date = datetime.strptime(date_str, "%Y-%m-%d").date()
        except ValueError:
            # 형식이 다르면 그대로 문자열만 사용
            self.date = None

    def __str__(self):
        # date가 파싱됐으면 예쁘게, 아니면 원래 문자열 출력
        if self.date is not None:
            return f"[{self.date}] {self.text}"
        else:
            return f"[{self.date_str}] {self.text}"


# 2) CSV 파일 읽어서 Record 객체 리스트에 저장
records = []

# data.csv 파일이 현재 파이썬 실행 파일(또는 노트북)과 같은 폴더에 있다고 가정
file_path = "data/data.csv"

with open(file_path, "r", encoding="utf-8") as f:
    reader = csv.reader(f)

    # 첫 줄이 헤더: date,text 인 경우 헤더 건너뛰기
    header = next(reader, None)
    # header가 실제로 'date','text'인지 확인해봐도 됨 (선택)

    for row in reader:
        if len(row) < 2:
            continue  # 열이 부족한 줄 건너뜀

        date_str, text = row[0], row[1]
        record = Record(date_str, text)
        records.append(record)

# 3) 클래스 객체들 화면에 출력
print("=== data.csv 레코드 출력 ===")
for rec in records:
    print(rec)


=== data.csv 레코드 출력 ===
[2025-09-24] 디지털 성범죄 처벌 강화 법안이 국회에 제출되어 처벌 수위 논의가 진행된다. 市는 후속 대책을 검토 중이라고 설명했다. 보도 여론은 진전 반응이 확산되는 양상이다
[2025-09-19] 온라인 커뮤니티의 혐오표현 규제와 표현의 자유 균형을 두고 공청회가 열렸다. 전문가들은 추가 보완이 필요하다고 밝혔다. 보도 여론은 피해 반응이 확산되는 양상이다
[2025-09-13] 청년 주거 대책으로 월세 지원과 공공주택 공급 확대 방안이 발표됐다. 정부는 관계부처 합동 점검을 실시한다고 밝혔다. 보도 여론은 만족 반응이 확산되는 양상이다
[2025-09-21] 대형 쇼핑몰 개인정보 유출 의혹이 제기되어 금융당국이 조사에 착수했다. 전문가들은 추가 보완이 필요하다고 밝혔다. 포럼 여론은 칭찬 반응이 확산되는 양상이다
[2025-09-28] 플랫폼 배달 수수료 인하와 소상공인 상생 협약이 지역별로 확산된다. 市는 후속 대책을 검토 중이라고 설명했다. 포럼 여론은 혜택 반응이 확산되는 양상이다
[2025-09-13] 대형 쇼핑몰 개인정보 유출 의혹이 제기되어 금융당국이 조사에 착수했다. 시민단체는 공개 토론을 제안했다. 뉴스 여론은 항의 반응이 확산되는 양상이다
[2025-09-20] 자율주행 차량 안전 기준과 보험 제도 정비를 위한 시범사업이 확대된다. 시민단체는 공개 토론을 제안했다. 포럼 여론은 실패 반응이 확산되는 양상이다
[2025-09-19] 대형 쇼핑몰 개인정보 유출 의혹이 제기되어 금융당국이 조사에 착수했다. 정부는 관계부처 합동 점검을 실시한다고 밝혔다. 보도 여론은 혜택 반응이 확산되는 양상이다
[2025-09-16] 자율주행 차량 안전 기준과 보험 제도 정비를 위한 시범사업이 확대된다. 정부는 관계부처 합동 점검을 실시한다고 밝혔다. 뉴스 여론은 혜택 반응이 확산되는 양상이다
[2025-09-07] 온라인 커뮤니티의 혐오표현 규제와 표현의 자유 균형을 두고 공청회가 열렸다. 정부는 관계부처 

### 13장 파일 실습 과제2
- 2000.csv 파일 읽어 클래스 객체로 만들기

In [66]:
import csv
from datetime import datetime

# 1) 레코드를 담을 클래스 정의
class Record:
    def __init__(self, date_str, text):
        # date_str: "2021-01-01" 같은 문자열
        self.date_str = date_str
        self.text = text

        # 날짜를 datetime 객체로도 저장해두고 싶으면:
        try:
            self.date = datetime.strptime(date_str, "%Y%m%d%H%M%S")
        except ValueError:
            # 형식이 다르면 그대로 문자열만 사용
            self.date = None

    def __str__(self):
        # date가 파싱됐으면 예쁘게, 아니면 원래 문자열 출력
        if self.date is not None:
            return f"[{self.date.strftime('%Y-%m-%d, %H:%M:%S')}] {self.text}"
        else:
            return f"[{self.date_str}] {self.text}"


# 2) CSV 파일 읽어서 Record 객체 리스트에 저장
records = []

# data.csv 파일이 현재 파이썬 실행 파일(또는 노트북)과 같은 폴더에 있다고 가정
file_path = "data/2000.csv"

with open(file_path, "r", encoding="utf-8") as f:
    reader = csv.reader(f)

    # 첫 줄이 헤더: date,text 인 경우 헤더 건너뛰기
    header = next(reader, None)
    # header가 실제로 'date','text'인지 확인해봐도 됨 (선택)

    for row in reader:
        if len(row) < 2:
            continue  # 열이 부족한 줄 건너뜀

        date_str, text = row[1], row[3]
        record = Record(date_str, text)
        records.append(record)

# 3) 클래스 객체들 화면에 출력
print("=== 2000.csv 레코드 출력 ===")
for rec in records:
    print(rec)


=== 2000.csv 레코드 출력 ===
[2025-07-01, 18:55:00] 한성숙 중기장관 후보자, 180억원대 재산 신고…스톡옵션 등 포함 땐 440억원대
[2025-03-18, 17:38:00] 헌재, 朴법무 탄핵심판 첫 변론으로 종결
[2025-05-16, 14:25:00] 게임 핵 팔아 1억4000만원 챙긴 남성…대법 "범죄수익 추징해야"
[2025-03-04, 18:31:00] 개인정보위, 로보락 등 로봇청소기 개인정보 수집 실태점검
[2025-07-11, 14:51:00] 경찰, 의료계 커뮤니티 `메디스태프` 복귀 전공의·의대생 협박글 작성자 추적
[2025-05-28, 09:08:00] "저열한 혐오정치..귀를 의심"..이준석 `성폭력 발언`에 고발 예고
[2025-05-30, 07:00:00] 결혼식에 경호원까지…교사 10명 고소·100번 넘게 민원 [취재후]
[2025-04-15, 16:29:00] "비밀번호보다 안전”… 카카오페이 암호화 논란에 전문가 반박
[2025-01-23, 10:32:00] `서부지법 난동자 53명` 신상박제 사이트 등장…尹 지지 유명인도 등재
[2025-01-05, 10:59:00] 대법원 "이태원 참사 희생자 성적 비하, 음란물 유포 해당"
[2025-01-10, 19:48:15] 빌리프랩 측 “민희진 표절 주장, 아일릿 좌표찍기”..20억 손배소 돌입 [종합]
[2025-04-21, 18:13:00] 민주 "홍준표, 불법 여론조사 의혹 연루"…洪 "거짓 공작 중단하라"
[2025-02-17, 11:13:00] 딥시크 신규가입 못한다…"국내 서비스 잠정 중단"
[2025-03-19, 10:52:00] “배우자 외도 담긴 CCTV, 개인정보라며 안 주는 호텔”… 방법은 [여행 팩트체크]
[2025-01-14, 16:01:00] 방통위, 한국판 DSA법 제정 추진…단통법 후속조치 나온다
[2025-02-13, 00:01:00] "하늘이 살해 교사, 전교조 출신에 정신병원 입원했대"…혐오·가짜뉴스 `확산`
[20